src/shell/dotnet/Shell/MainWindow.xaml.cs

```diff
@@ -107,6 +106,15 @@ public ModuleViewModel(IModuleManifest manifest, ImageSourceProvider imageSource
                         webManifestDetails.Url);
                 }
             }
+            else if (manifest.TryGetDetails<NativeManifestDetails>(out var nativeManifestDetails))
+            {
+                var icon = System.Drawing.Icon.ExtractAssociatedIcon(Path.GetFullPath(nativeManifestDetails.Path.ToString()));
```

Comment:
`// TODO: Let the manifest specify the icon`

Reply:
I added a property to the nativemanifestdetails, but it wouldn't work with the current implementation of imagesourceprovider, so I'm not adding it here yet.